src/JSDomEnvironment.js

```diff
@@ -104,8 +107,17 @@ JSDomEnvironment.prototype.dispose = function() {
   //this.global.close();
 };
 
+/**
+ * Evaluates the given source text as if it were in a file with the given name
+ * and returns the result.
+ */
 JSDomEnvironment.prototype.runSourceText = function(sourceText, fileName) {
-  return this.global.run(sourceText, fileName);
+  // TODO: Stop using the private API and instead configure jsdom with a
+  // resource loader and insert <script src="${filename}"> in the document. The
+  // reason we use vm for now is because the script element technique is slow.
+  return vm.runInContext(sourceText, this.document._ownerDocument._global, {
```

Comment:
Yeah, would be nice having it exposed as public API (e.g. getter only for the global object, or something, @Sebmaster, @domenic). Also, this implementation details make it less clear in the code (for those who will be dealing with it later) why we use `this.document._ownerDocument._global`, but also have e.g. `this.global` (which is `this.document.defaultView`). So worth adding a comment about it as well.